angularjs/manageVariable/edit.controller.js

```diff
@@ -258,13 +258,35 @@
             return true;
         }
 
+        function validateNameField() {
+            var regex = /^[a-zA-Z0-9\s]+$/;
```

Comment:
That actually does not do the same as the validator on PHP side, is that on purpose? It won't allow foreign languages and also white space characters that are not a simple space.

Reply:
yeah I know, I wanted to look into this but forgot. The PHP version is much better, I will "translate" that into JS.

Reply:
Hi @sgiehl,
I can't use the same, unicode version of the regex as in PHP, because IE does not support it.
Should we leave the front-end validation? Or should I try to write a regex without unicode?

We still support IE11, don't we?

Reply:
FYI, I tried [babel](https://babeljs.io/repl#?browsers=defaults%2C%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&spec=false&loose=false&code_lz=DYUwLgBATiDmIA8IF4IHoB6BtAOgBwG8AZAX3wIFkzCA5EiAXQGoASNAVwChRIBnMKCggByAIyjhnTjHgIAdGBD8AFPygBKKZyA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2&prettier=false&targets=&version=7.13.15&externalPlugins=), and got a very long regex, and still not sure it supports every language.

I would probably leave the front-end validation as I can't hide that line from IE. Or is there a way to "hide" it?

Reply:
In that case it might be better not to validate here at all and let the backend return the error message if it doesn't match.
Otherwise we should adjust the validation in backend to match the one in javascript.
Not sure if foreign languages should be allowed in names at all. ping @tsteur